app/src/main/java/com/cornellappdev/android/eateryblue/ui/screens/EateryDetailScreen.kt

```diff
@@ -60,58 +60,62 @@ fun EateryDetailScreen(
     val coroutineScope = rememberCoroutineScope()
     var issue by remember { mutableStateOf<Issue?>(null) }
 
-    ModalBottomSheetLayout(
-        sheetState = modalBottomSheetState,
-        sheetContent = {
-            when (sheetContent) {
-                BottomSheetContent.PAYMENT_METHODS_AVAILABLE -> {
-                    PaymentMethodsAvailable(selectedPaymentMethods = paymentMethods) {
-                        coroutineScope.launch {
-                            modalBottomSheetState.hide()
+    when (val eateryApiResponse = eateryDetailViewModel.eatery.collectAsState().value) {
+        is EateryApiResponse.Pending -> {
+            EateryDetailLoadingScreen(shimmer)
+        }
+
+        is EateryApiResponse.Error -> {
+            Text(text = "ERROR")
+        }
+
+        is EateryApiResponse.Success -> {
+            val eatery = eateryApiResponse.data
+            ModalBottomSheetLayout(
+                sheetState = modalBottomSheetState,
+                sheetContent = {
+                    when (sheetContent) {
+                        BottomSheetContent.PAYMENT_METHODS_AVAILABLE -> {
+                            PaymentMethodsAvailable(selectedPaymentMethods = paymentMethods) {
+                                coroutineScope.launch {
+                                    modalBottomSheetState.hide()
+                                }
+                            }
                         }
-                    }
-                }
-                BottomSheetContent.HOURS -> {
-                    // TODO finish
-                }
-                BottomSheetContent.WAIT_TIME -> {
-                    // TODO finish
-                }
-                BottomSheetContent.REPORT -> {
-                    eateryDetailViewModel.eatery.id?.let {
-                        ReportBottomSheet(
-                            issue = issue,
-                            eateryid = it,
-                            sendReport = { issue, report, eateryid ->
-                                eateryDetailViewModel.sendReport(issue, report, eateryid)
-                            }) {
-                            coroutineScope.launch {
-                                modalBottomSheetState.hide()
+
+                        BottomSheetContent.HOURS -> {
+                            // TODO finish
+                        }
+
+                        BottomSheetContent.WAIT_TIME -> {
+                            // TODO finish
+                        }
+
+                        BottomSheetContent.REPORT -> {
+                            eatery.id?.let {
```

Comment:
Why is `id` nullable here? Shouldn't this be a non-nullable field since we are in the success state of the query?

Reply:
I'm really not quite sure, but that's how the `Eatery` models have always been. It might be due to the fact that we're parsing Eatery objects from backend's JSON, and backend may be making all the fields nullable (or something to do with JSON needs them to be nullable?)... it's also very possible though that this is unnecessary, and we can make some of the `Eatery` fields non-null. It certainly would clean some things up.

![image](https://github.com/cuappdev/eatery-blue-android/assets/47724806/8ff945b6-ddfd-48b6-8d2c-8b53a16148ac)

Reply:
wait it is actually disgusting that the entire class is nullable lmao, it should definitely be refactored to make all fields non-null. This would make everything way safer and more concise. Maybe not for this PR but it definitely needs to happen before launch.